python/lsst/pex/config/config.py

```diff
@@ -64,6 +64,8 @@ def _autocast(x, dtype):
         return float(x)
     if dtype == int and isinstance(x, long):
         return int(x)
+    if isinstance(x, str):
```

Comment:
As I say in lsst/afw#83 the problem is that I can't promise that people outside are giving me `str`. The reason this exists is because tests started failing in `afw` and I can imagine code further up the stack also not doing it consistently. That and the fact that string literals are not future `str` leads me to believe this is the only option. Casting to a proper `str` is probably the best thing anyhow.